module/VuFind/src/VuFind/Mailer/Mailer.php

```diff
@@ -180,83 +160,91 @@ public function setTransport($transport)
      *
      * @param string $input String to convert
      *
-     * @return AddressList
+     * @return array
      */
-    public function stringToAddressList($input)
+    public function stringToAddressList($input): array
     {
         // Create recipient list
-        $list = new AddressList();
+        $list = [];
         foreach (preg_split('/[\s,;]/', $input) as $current) {
             $current = trim($current);
             if (!empty($current)) {
-                $list->add($current);
+                $list[] = $current;
             }
         }
         return $list;
     }
 
     /**
-     * Constructs a {@see MimeMessage} body from given text and html content.
+     * Constructs an {@see Email} body from given text and html content.
      *
      * @param string|null $text Mail content used for plain text part
      * @param string|null $html Mail content used for html part
      *
-     * @return MimeMessage
+     * @return Email
      */
     public function buildMultipartBody(
         string $text = null,
         string $html = null
-    ): MimeMessage {
-        $parts = new MimeMessage();
-
-        if ($text) {
-            $textPart = new MimePart($text);
-            $textPart->setType(Mime::TYPE_TEXT);
-            $textPart->setCharset('utf-8');
-            $textPart->setEncoding(Mime::ENCODING_QUOTEDPRINTABLE);
-            $parts->addPart($textPart);
+    ): Email {
+        $email = $this->getNewMessage();
+        if (null !== $text) {
+            $email->text($text);
         }
-
-        if ($html) {
-            $htmlPart = new MimePart($html);
-            $htmlPart->setType(Mime::TYPE_HTML);
-            $htmlPart->setCharset('utf-8');
-            $htmlPart->setEncoding(Mime::ENCODING_QUOTEDPRINTABLE);
-            $parts->addPart($htmlPart);
+        if (null !== $html) {
+            $email->html($html);
         }
-
-        $alternativePart = new MimePart($parts->generateMessage());
-        $alternativePart->setType('multipart/alternative');
-        $alternativePart->setBoundary($parts->getMime()->boundary());
-        $alternativePart->setCharset('utf-8');
-
-        $body = new MimeMessage();
-        $body->setParts([$alternativePart]);
-
-        return $body;
+        return $email;
     }
 
     /**
      * Send an email message.
      *
-     * @param string|Address|AddressList $to      Recipient email address (or
-     * delimited list)
-     * @param string|Address             $from    Sender name and email address
-     * @param string                     $subject Subject line for message
-     * @param string|MimeMessage         $body    Message body
-     * @param string                     $cc      CC recipient (null for none)
-     * @param string|Address|AddressList $replyTo Reply-To address (or delimited
-     * list, null for none)
+     * @param string|Address|Address[]      $to      Recipient email address(es) (or delimited list)
```

Comment:
Should this and other relevant parameters here be `string|string[]|Address|Address[]` to be completely accurate?

Reply:
Good point. Fixed!